opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -1345,6 +1346,14 @@ namespace Opm {
 
         // We use the rates from the privious time-step to reduce oscilations
         WellGroupHelpers::updateWellRates(fieldGroup, schedule(), reportStepIdx, previous_well_state_, well_state_);
+
+        // Set ALQ for off-process wells to zero
+        for (const auto& well : schedule().getWells(reportStepIdx)) {
+            if (well.isProducer() && well_state_.wellMap().count(well.name()) == 0) {
+                well_state_.setALQ(well.name(), 0.0);
+            }
+        }
```

Comment:
You can still do this in terms of `schedule().wellNames()` instead of `schedule().getWells()` if you don't want to copy complete well objects:
```C++
for (const auto& wname : schedule().wellNames(reportStepIdx)) {
    if (schedule().getWell(wname, reportStepIdx).isProducer() &&
        (well_state_.wellMap().count(wname) == 0))
    {
        well_state_.setALQ(wname, 0.0);
    }
}
```

Reply:
I see, this uses the Well reference. I guess there will be one more round...